lib/storage/s3.js

```diff
@@ -0,0 +1,146 @@
+"use strict";
+
+class S3 {
+    /**
+     * @param {Object} awsSdk
+     * @param {Object} promise
+     */
+    constructor(awsSdk, promise) {
+        this.aws = awsSdk;
+        this.config = {};
+        this.configure({});
+        this.promise = promise;
+        this.s3Bucket = null;
+    }
+
+
+    /**
+     * Creates the S3 bucket we need for subsequent calls.
+     *
+     * @return {Object} s3Bucket
+     */
+    transit() {
+        if (! this.s3Bucket) {
+            this.s3Bucket = new this.aws.S3({params: {Bucket: this.config.bucket}});
+
+            this.s3Bucket = this.promise.promisifyAll(this.s3Bucket);
+        }
+
+        return this.s3Bucket;
+    }
+
+
+    /**
+     * Configures options for AWS.
+     *
+     * @param {Object} config
+     * @return {Object} this
+     */
+    configure(config) {
+        this.aws.config.region = config.region || "us-east-1";
+        this.config.bucket = config.bucket || null;
+        config = null;
```

Comment:
Unnecessary line, unless you're intentionally freeing up some large amount of memory.